test/fx/test_gradual_type.py

```diff
@@ -330,6 +335,76 @@ def forward(self, x: Dyn):
         with self.assertRaises(TypeError):
             tc.type_check()
 
+    def test_type_check_conv2D(self):
```

Comment:
I think a couple things would also make the tests more robust:

1) Can we test that the inferred sizes match with the sizes we actually see when running the module?
2) Can we make the testing more exhaustive? e.g. automate testing over different `kernel_size`, `stride`, `padding`, `dilation`, `groups` parameters and make sure it works across all those parameters

This PR might take a while to get fully tested/correct so it might be a good idea to pull it out from the stack and develop it separately